lib/chef/provisioning/aws_driver/driver.rb

```diff
@@ -644,6 +644,10 @@ def cloudsearch(api_version="20130101")
       @cloudsearch[api_version]
     end
 
+    def elasticsearch
+      @elasticsearch ||= Aws::ElasticsearchService::Client.new(region: aws_config.region)
```

Comment:
This should follow the pattern from https://github.com/chef/chef-provisioning-aws/blob/994d25daa9912daaf991f41966670b714775541a/lib/chef/provisioning/aws_driver/driver.rb#L655 and just pass the entire `aws_config_2` hash to the client.

Actually, will you just change [`AWS_V2_SERVICES.each`](https://github.com/chef/chef-provisioning-aws/blob/994d25daa9912daaf991f41966670b714775541a/lib/chef/provisioning/aws_driver/driver.rb#L651) to look like

```ruby
    AWS_V2_SERVICES.each do |load_name, short_name|
      class_eval <<-META
      def #{short_name}_client
        @#{short_name}_client ||= ::Aws::#{load_name}::Client.new(**aws_config_2)
      end
      META
      
      if defined? ::Aws::#{load_name}::Resource
      class_eval <<-META
        def #{short_name}_resource
          @#{short_name}_resource ||= ::Aws::#{load_name}::Resource.new(**(aws_config_2.merge({client: #{short_name}_client})))
        end
      META
```

and add `"ElasticsearchService" => "elasticsearch"` to the [`AWS_V2_SERVICES`](https://github.com/chef/chef-provisioning-aws/blob/994d25daa9912daaf991f41966670b714775541a/lib/chef/provisioning/aws_driver/driver.rb#L31) hash then you don't need to define this here